pytorch_lightning/core/hooks.py

```diff
@@ -680,18 +680,6 @@ def predict_dataloader(self) -> EVAL_DATALOADERS:
             will have an argument ``dataloader_idx`` which matches the order here.
         """
 
-    def on_train_dataloader(self) -> None:
-        """Called before requesting the train dataloader."""
-
-    def on_val_dataloader(self) -> None:
-        """Called before requesting the val dataloader."""
-
-    def on_test_dataloader(self) -> None:
-        """Called before requesting the test dataloader."""
-
-    def on_predict_dataloader(self) -> None:
-        """Called before requesting the predict dataloader."""
-
```

Comment:
this is a removal, not a deprecation. given that these are on the stable API, we need to mark these methods as deprecated in v1.5 and that they will be removed in v1.7. it has to be a 2-stage process to allow end users a chance to migrate before the breaking change happens.

Reply:
good point! I will keep them for now. Should I just update the docs and maybe add `rank_zero_deprecation` for the update? Similar to deprecating `running_sanity_check`?

Reply:
- Let's update the docstrings to reflect that these are deprecated
- We could add a utility function like https://github.com/PyTorchLightning/pytorch-lightning/blob/de57feff445744f431eaef29182b1bcbf58a1924/pytorch_lightning/trainer/configuration_validator.py#L22 

which accepts the lightning module and datamodule as inputs and outputs a deprecation warning if any deprecated hooks are overridden. @Borda @awaelchli what do you think?